lib/cloud/aws/policy_statements.go

```diff
@@ -188,8 +189,8 @@ func StatementForEKSAccess() *Statement {
 
 // StatementForAWSOIDCRoleTrustRelationship returns the Trust Relationship to allow the OpenID Connect Provider
 // set up during the AWS OIDC Onboarding to assume this Role.
-func StatementForAWSOIDCRoleTrustRelationship(accountID, providerURL string, audiences []string) *Statement {
-	federatedARN := fmt.Sprintf("arn:aws:iam::%s:oidc-provider/%s", accountID, providerURL)
+func StatementForAWSOIDCRoleTrustRelationship(partition, accountID, providerURL string, audiences []string) *Statement {
```

Comment:
Changing this method requires a change to
https://github.com/gravitational/teleport/blob/d3025b2a282cf80edb154e987ef8bc05f232054c/lib/integrations/awsoidc/idp_iam_config.go#L279

Did you test this?